src/pages/index.js

```diff
@@ -50,38 +88,88 @@ addImagePopup.setEventListeners();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardsList = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      createCard(item);
-    },
-  },
-  cardsListSection
-);
-cardsList.renderItems();
+const confirmPopup = new PopupConfirm("#confirm-modal", (cardData, evt) => {
+  evt.preventDefault();
+  deleteCard(cardData);
+});
+confirmPopup.setEventListeners();
 
 //add event listeners
-profileEditButton.addEventListener("click", openProfileForm);
-profileAddButton.addEventListener("click", openAddForm);
-
-//event listener callbacks
-function openProfileForm() {
+profileEditButton.addEventListener("click", () => {
   profilePopup.open();
-  const { name, title } = userInfo.getUserInfo();
-  profileModalNameInput.value = name;
-  profileModalDescInput.value = title;
+  profileModalNameInput.value = userInfo.name.textContent;
+  profileModalDescInput.value = userInfo.about.textContent;
   profileFormValidator.resetValidation();
   profileFormValidator.toggleButtonState();
+});
+profileAddButton.addEventListener("click", () => addImagePopup.open());
+profileAvatarEdit.addEventListener("click", () => avatarEditPopup.open());
+
+//functions
+function createCard(card) {
+  const newCard = new Card(
+    card,
+    "#card-template",
+    () => {
+      imagePopup.open(newCard);
+    },
+    (cardData) => {
+      confirmPopup.open(cardData);
+    },
+    (cardData) => {
+      updateCardLike(cardData);
+    }
+  );
+  return newCard.generateCardElement();
 }
 
-function openAddForm() {
-  addImagePopup.open();
+function deleteCard(cardData) {
+  const deleteCardApi = new Api({
+    url: `${apiData.cards}/${cardData.cardId}`,
+    method: "DELETE",
+    headers: apiData.headers,
+  });
+  deleteCardApi.handleFetch();
+  cardData.cardEl.remove();
+  cardData.cardEl = null;
 }
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.